### ISSUE TITLE
backpropergation for operation with custom jvp cannot use 

### ISSUE THREAD

**Issue by FranzAtGithub** [google/jax#12158]
Hey there,

I tried to circumvent the problem in [12142](https://github.com/google/jax/issues/12142) by defining a new operation `my_abs` with a custom jvp rule like this:

```python
import jax
import jax.numpy as jnp


@jax.custom_jvp
def my_abs(x):
    return jnp.abs(x)


@my_abs.defjvp
def my_abs_jvp(primals, tangents):
    x, = primals
    d, = tangents

    # this should be correct but does not work
    return jnp.abs(x), jnp.where(x > 0, d, jnp.where(x < 0, -d, jnp.abs(d)))

    # this works but is incorrect
    # return jnp.abs(x), jnp.where(x > 0, d, jnp.where(x < 0, -d, 2*d))

    # this does also not work but it is also incorrect
    # return jnp.abs(x), jnp.where(x > 0, d, jnp.where(x < 0, -d, jnp.sin(d)))
```

This works as expected as can be seen here:
```python
import itertools
xs = [-1., 0., 1.]
ds = [-1., 1.]
for x, d in itertools.product(xs, ds):
    print(f"abs'({x}; {d}) = {jax.jvp(my_abs, [x], [d])[1]}")
```

As discussed also in [10515](https://github.com/google/jax/issues/10515) the gradient for `abs` is of course not well defined. One has to choose any value in the interval `[-1, 1]`. This is what backpropergation for `jnp.abs` does. Sadly, my own `my_abs` definition does not work for backpropergration as this snipped shows:
```python
# this has the correct jvp but grad does not work
grad_abs = jax.grad(my_abs)

# this does work but has the wrong jvp
# grad_abs = jax.grad(jnp.abs)
for x in xs:
    print(f"∇abs({x}) = {grad_abs(x)}")
```
I get a `NotImplementedError`. The problem disappears when I use the second (wrong and totally made up) jvp rule
```python
jnp.abs(x), jnp.where(x > 0, d, jnp.where(x < 0, -d, 2*d))
```
For the third option I again get the `NotImplementedError`. This is somehow reasonable as JVPs (directional derivatives) are always linear w.r.t. the tangents for differentiable functions so probably there was no need for `abs` and `sin` to be valid in this situation.

For me I would just like to keep the original behavior of `jnp.abs` in terms of backwards propergation but my new custom jvp for forward mode. Or a fix of [12142](https://github.com/google/jax/issues/12142) ^^

Ty for reading,
Franz
